### PR TITLE
Add environment variable to control in-memory storage span limit

### PR DESCRIPTION
**Pull request by anuraaga**
Regardless of whether we merge https://github.com/openzipkin/docker-zipkin/pull/227 in it's probably still nice to add this knob.

### CODE REVIEW

**Review by devinsba — APPROVED** [review]

**Review by jeqo — APPROVED** [review]

File: zipkin-server/src/main/resources/zipkin-server-shared.yml [hunk 1]
```diff
@@ -82,7 +82,7 @@ zipkin:
       max-queue-size: ${STORAGE_THROTTLE_MAX_QUEUE_SIZE:1000}
     mem:
       # Maximum number of spans to keep in memory.  When exceeded, oldest traces (and their spans) will be purged.
-      max-spans: 500000
+      max-spans: ${STORAGE_MEM_MAX_SPANS:500000}
```
Comment (jorgheymans) [inline zipkin-server/src/main/resources/zipkin-server-shared.yml]: given that we have `CASSANDRA`, `ES` and `MYSQL` as prefixes for the other storage options maybe `INMEMORY_MAX_SPANS` ?

**Comment by jorgheymans** [comment]
When the name settles, would be good to note this config option in https://github.com/openzipkin/zipkin/tree/master/zipkin-server#in-memory-storage

**Comment by codefromthecrypt** [comment]
probably MEM is most intuitive as it matches the default storage type name.
only ES is exceptional in this naming convention iirc

**Comment by anuraaga** [comment]
Thanks - left the name and updated the docs

**Comment by codefromthecrypt** [comment]
I'll update and rename to MEM_MAX_SPANS

STORAGE_ is used for generic parameters at the moment.

File: zipkin-server/README.md [hunk 2]
```diff
@@ -163,9 +163,9 @@ Example usage:
 ```bash
 $ java -jar zipkin.jar
 ```
-You can override the maximum number of spans stored using the `--max-spans` application parameter:
+You can override the maximum number of spans stored using the `STORAGE_MEM_MAX_SPANS` environment variable:
 ```bash
-$ java -Xmx1G -jar zipkin.jar --max-spans=1000000
+$ STORAGE_MEM_MAX_SPANS=1000000 java -Xmx1G -jar zipkin.jar
```
Comment (codefromthecrypt) [inline zipkin-server/README.md]:
```suggestion
$ MEM_MAX_SPANS=1000000 java -Xmx1G -jar zipkin.jar
```

File: zipkin-server/src/main/resources/zipkin-server-shared.yml (hunk 1 above)
Comment (codefromthecrypt) [inline zipkin-server/src/main/resources/zipkin-server-shared.yml]:
```suggestion
      max-spans: ${MEM_MAX_SPANS:500000}
```

Comment (codefromthecrypt) [inline zipkin-server/src/main/resources/zipkin-server-shared.yml]:
```suggestion
      max-spans: ${MEM_MAX_SPANS:500000}
```

**Review by codefromthecrypt — COMMENTED** [review]
I'll raise a separate commit to add MEM_MAX_SPANS more explicitly to the readme

File: zipkin-server/src/main/resources/zipkin-server-shared.yml (hunk 1 above)
Comment (codefromthecrypt) [reply zipkin-server/src/main/resources/zipkin-server-shared.yml]: PTAL at last commit

**Review by codefromthecrypt — APPROVED** [review]
OK looks good to me!

**Review by anuraaga — COMMENTED** [review]
LGTM thanks for the edits

File: zipkin-server/README.md [hunk 3]
```diff
@@ -157,28 +157,26 @@ Defaults to true
...
 ```
-You can override the maximum number of spans stored using the `--max-spans` application parameter:
-```bash
-$ java -Xmx1G -jar zipkin.jar --max-spans=1000000
-```
 
-Note this storage component was primarily developed for testing and as a means to get Zipkin server up and running quickly without external dependencies. It is not viable for high work loads.  
+Note: this storage component was primarily developed for testing and as a means to get Zipkin server
+up and running quickly without external dependencies. It is not viable for high work loads. That
+said, ff you encounter out-of-memory errors, try decreasing `MEM_MAX_SPANS` or increasing the heap
```
Comment (codefromthecrypt) [inline zipkin-server/README.md]:
```suggestion
said, if you encounter out-of-memory errors, try decreasing `MEM_MAX_SPANS` or increasing the heap
```

File: zipkin-server/README.md [hunk 4]
```diff
@@ -157,28 +157,26 @@ Defaults to true
...
-$ java -Xmx1G -jar zipkin.jar --max-spans=1000000
-```
 
-Note this storage component was primarily developed for testing and as a means to get Zipkin server up and running quickly without external dependencies. It is not viable for high work loads.  
+Note: this storage component was primarily developed for testing and as a means to get Zipkin server
+up and running quickly without external dependencies. It is not viable for high work loads. That
+said, if you encounter out-of-memory errors, try decreasing `MEM_MAX_SPANS` or increasing the heap
+size (-Xmx).
 
-### Throttled Storage (Experimental)
```
Comment (codefromthecrypt) [inline zipkin-server/README.md:173]: I moved this down as noticed in the doc we actually say what's experimental before saying what's not

**Pull request merged**